src/main/java/tech/sourced/siva/SivaUnpacker.java

```diff
@@ -0,0 +1,135 @@
+package tech.sourced.siva;
+
+import org.apache.commons.io.IOUtils;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.attribute.PosixFilePermission;
+import java.util.List;
+import java.util.Set;
+import java.util.zip.CRC32;
+
+/**
+ * This is a high-level API for extracting bare git repository
+ * (content of .git) from a given .siva file.
+ * <p>
+ * Actual reading is done by SivaReader
+ *
+ * @see SivaReader#getEntry(IndexEntry)
+ */
+public class SivaUnpacker {
+    private SivaReader reader;
+    private Boolean ignorePerms = true;
+
+    public SivaUnpacker(SivaReader reader) {
+        this.reader = reader;
+    }
+
+    public SivaUnpacker(SivaReader reader, Boolean ignorePermissions) {
+        this(reader);
+        this.ignorePerms = ignorePermissions;
+    }
+
+    public SivaUnpacker(String sivaFilePath) throws SivaException {
+        try {
+            this.reader = new SivaReader(new File(sivaFilePath));
+        } catch (FileNotFoundException e) {
+            throw new SivaException("Could not read " + sivaFilePath, e);
+        }
+    }
+
+    public SivaUnpacker(String sivaFilePath, Boolean ignorePermissions) throws SivaException {
+        this(sivaFilePath);
+        this.ignorePerms = ignorePermissions;
+    }
+
+    public void unpack(String dstDir) throws SivaException {
+        File dst = new File(dstDir);
+        if (!dst.exists()) {
+            System.out.println("Directory " + dstDir + " does not exist, creating " + dst.getAbsolutePath());
```

Comment:
👍  I'll look into adding a logging that can be seamlessly used in Apache Spark, otherwise will just remove it.